patterns/java/src/refactoring_guru/patterns/singleton/example/non_therad_safe/Demo.java

```diff
@@ -0,0 +1,23 @@
+package refactoring_guru.patterns.singleton.example.non_therad_safe;
+
+public class Demo {
+    public static void main(String[] args) {
```

Comment:
How about creating two internal thread classes here (i.e. move ThreadDemo inside and create him a sibling with a "BAR" snippet's code)? It will improve obviousness of this example a lot when you see all thread classes together in one file.